lib/kubernetes-deploy/template_discovery.rb

```diff
@@ -1,14 +1,39 @@
 # frozen_string_literal: true
-
 module KubernetesDeploy
   class TemplateDiscovery
-    def initialize(template_dir)
-      @template_dir = template_dir
-    end
+    class << self
+      def templates(template_args)
+        template_paths = {}
+        dir_args = template_args.select { |arg| File.directory?(arg) }
+        file_args = template_args.select { |arg| File.file?(arg) }
+
+        # Directory arg
+        dir_args.each_with_object(template_paths) do |template_dir, hash|
+          hash[template_dir] = Dir.foreach(template_dir).select do |filename|
+            filename.end_with?(".yml.erb", ".yml", ".yaml", ".yaml.erb")
+          end
+        end
+        # Filename arg
+        file_args.each_with_object(template_paths) do |filename, hash|
+          dir_name = File.dirname(filename)
+          hash[dir_name] ||= []
+          hash[dir_name] << File.basename(filename) unless hash[dir_name].include?(filename)
+        end
+
+        template_paths
```

Comment:
Now any task should be able to call `TemplateDiscovery#templates` and receive back a mapping of `dirname: [files]`. The input of `templates` is an array whose elements are either file paths or directories (abs and/or relative). The nice thing here is that there's zero other dependencies for generating the dir->files mappings